cli/lib/tokens.mjs

```diff
@@ -13,7 +13,15 @@ Sends an approve transaction to an ERC20/ERC721/ERC1155 contract for a certain a
 * @param {object} provider  - web3 provider
 * @returns {Promise} transaction
 */
-async function approve(ercAddress, ownerAddress, spenderAddress, tokenType, value, provider) {
+async function approve(
+  ercAddress,
+  ownerAddress,
+  spenderAddress,
+  tokenType,
+  value,
+  provider,
+  encodeABI,
```

Comment:
This must be `ethereumSigningKey` not `encodeABI`

Reply:
No, Wallet doesn’t have access to signing key. `encodeABI` is a flag to let the approve function be sent and signed by metamask

Reply:
Should this `ethereumSigningKeyExists` because we seem to be checking with condition `!!ethereumSigningKeyExists` in the function that calls it?

Reply:
For the wallet case `!!ethereumSigningKey` is false, so approve will be executed as
 ``` await ercContract.methods
          .approve(spenderAddress, APPROVE_AMOUNT)
          .send({ from: ownerAddress });
```